File: Python/refer-offer/nge_tou_zi_de_dian_shu_lcof.py
from typing import *

def dicesProbability(n: int) -> List[float]:
    dp = [1 / 6] * 6
    for i in range(2, n + 1):
        temp = [0] * (5 * i + 1)
        for j in range(len(dp)):
            for k in range(6):
                temp[j + k] += dp[j] / 6
        dp = temp
    return dp

File: Java/src/com/dbc/algorithm/ReferOffer/NgeTouZiDeDianShuLcof.java
package com.dbc.algorithm.ReferOffer;

import java.util.Arrays;

public class NgeTouZiDeDianShuLcof {
    public double[] dicesProbability(int n) {
        double[] res = new double[6];
        Arrays.fill(res, 1. / 6.);
        for (int i = 2; i <= n; i++) {
            double[] temp = new double[5 * i + 1];
            for (int j = 0; j < res.length; j++) {
                for (int k = 0; k < 6; k++) {
                    temp[j + k] += res[j] / 6.;
                }
            }
            res = temp;
        }
        return res;
    }
}
